src/cached_object_store/lru.rs

```diff
@@ -0,0 +1,502 @@
+#![allow(clippy::needless_lifetimes)]
+#![allow(clippy::extra_unused_lifetimes)]
+#![allow(clippy::unwrap_used)]
+#![allow(unused)]
+//! An implementation of a generic LRU cache. It supports `get`, `put`, `remove`,
+//! and `remove_entry`, etc. Allow us to hold multiple immutable references to values in LruCache.
+//! Also support dynamical adjustment of the capacity and size aware eviction like moka and foyer.
+//! For the use of hold multiple immutable references to values at the same time, see unit tests.
+//! Heavily influenced by the [Design safe collection API with compile-time reference stability in Rust]
+//! (<https://blog.cocl2.com/posts/rust-ref-stable-collection/>)
+
+// SipHash is too slow, even rustc does not choose it, and we don't need to worry about DoS attacks
+// Details: https://github.com/rust-lang/rustc-hash
+use hashbrown::HashMap;
```

Comment:
isn't this implementation the same as the standard library? From the [crate's readme](https://crates.io/crates/hashbrown):

> Since Rust 1.36, this is now the HashMap implementation for the Rust standard library. However you may still want to use this crate instead since it works in environments without std, such as embedded systems and kernels.

Reply:
std uses SipHash

Reply:
what is the performance difference? It'd be good to have data before deciding to replace the standard hashmap with an additional crate to make a more informed decision on whether this is necessary or not.

Reply:
https://github.com/orlp/foldhash

https://github.com/rust-lang/rustc-hash

Reply:
i suppose this comment may be solved as this pr have cleaned up the checked-in lru code